module/zcommon/zfs_namecheck.c

```diff
@@ -183,6 +183,8 @@ entity_namecheck(const char *path, namecheck_err_t *why, char *what)
 {
 	const char *end;
 
+	ASSERT((why == NULL && what == NULL) || (why != NULL && what != NULL));
```

Comment:
You can use `EQUIV(why == NULL, what == NULL)` for this, and it'll give a better error message